sympy/geometry/polygon.py

```diff
@@ -81,16 +81,20 @@ class Polygon(GeometrySet):
     >>> Polygon(p1, p2, p5)
     Segment2D(Point2D(0, 0), Point2D(3, 0))
 
-    While the sides of a polygon are not allowed to cross implicitly, they
-    can do so explicitly. For example, a polygon shaped like a Z with the top
-    left connecting to the bottom right of the Z must have the point in the
-    middle of the Z explicitly given:
+    Both implicit and explicit intersections are allowed. For example,
+    consider a polygon shaped like a Z with the top left connecting to the
+    bottom right of the Z. The point in the middle of the Z may or may not be
+    explicitly given. However it should be noted that not all characteristics
+    of the implicit case be accurate(such as area):
 
```

Comment:
Suggested rewrite to the effect of:

```
When a segment connecting two of the provided vertices crosses
one or more other segments of the polygon, the points of intersection
will be automatically inserted into the arguments. Note that two
polygons may have the same shape but have different area since area
depends on the *path* taken to create the polygon. For example, 
creating a Z with a line connecting the bottom right to the top left
gives an overall shape of two triangles meeting at common vertex. 
Defining/drawing the shape this way, however, is consistent with
moving in a counter-clockwise direction around the lower triangle and 
a clockwise direction around the top triangle so their areas cancel:

>>> mid, tl, tr, br, bl = (0,0), (-1,1), (1,1), (1,-1), (-1, -1)
>>> cwccw = Polygon(mid, tl, tr, mid, bl, br, mid)
>>> cwccw.area
0

If the same shape is made by defining a clockwise path around
both triangles, the areas do not cancel:

>>> cw = Polygon(mid, tl, tr, mid, br, bl, mid)
>>> cw.area
-2

When allowing the intersections to be computed automatically,
the path is consistent with the path specified by the vertices:

>>> Polygon(tl, tr, bl, br, tl).area
0

To disallow intersections...[blah blah]...then an error will be raised.  <-- is that right?
```

The caution sounds a little scary in saying that the area may not be accurate.